test/k8s-local-cluster-test/provision-eks-sqs-infra

```diff
@@ -0,0 +1,386 @@
+            -r          Region (default us-east-2)
+
+          Optional:
+            -a          Account ID
+            -s          SQS Queue Name
+            -f          Add a Spot Fleet Node Group
+            -d          Delete infrastructure
+            -v          Verbose output
+            -h          Show this usage help
+EOM
+)
+
+usage() {
+  echo "$USAGE" 1>&2
+  exit "$1"
+}
+
+# Process our input arguments
+while getopts "hn:r:a:s:vdf" opt; do
+  case ${opt} in
+    a ) # Account ID
+        ACCOUNT_ID="${OPTARG}"
+      ;;
+    n ) # Name
+        CLUSTER_NAME="${OPTARG}"
+      ;;
+    s ) # SQS Queue Name?
+        SQS_QUEUE_NAME="${OPTARG}"
+      ;;
+    r ) # AWS Region
+        AWS_REGION="${OPTARG}"
+      ;;
+    d ) # Delete infa
+        DELETE=1
+      ;;
+    f ) # Spot Fleet for ITN testing
+        FLEET=1
+      ;;
+    v ) # VERBOSE
+        set -x
+      ;;
+    h ) # HELP
+        usage 0
+      ;;
+    \? )
+        usage 1
+      ;;
+  esac
+done
+
+if [[ -n "${ENDPOINT_URL}" ]]; then
+    EXTRA_AWS_PARAMS=" ${ENDPOINT_URL}"
+fi
+
+if [[ -z "${AWS_REGION}" ]]; then
+    echo "You must specify a region w/ -r"
+    exit 1
+fi
+
+if [[ -z "${CLUSTER_NAME}" ]]; then
+    echo "You must specify the cluster name w/ -n"
+    exit 1
+fi
+
+ECR_PLAYGROUND_REPO="${ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com"
+POD_ROLE="cth-${CLUSTER_NAME}-role"
+
+ACTUAL_ACCOUNT_ID=$(aws sts get-caller-identity | jq -r .Account)
+if [[ "${ACTUAL_ACCOUNT_ID}" != "${ACCOUNT_ID}" ]]; then
+  echo "Credentials are for ${ACTUAL_ACCOUNT_ID} but you specified ${ACCOUNT_ID}"
+  echo "Setup credentials for ${ACCOUNT_ID} or correct the -a argument"
+  exit 1
+fi
+
+echo "======================================================================"
+echo "Using the following parameters: "
+echo "                    AWS Account ID: ${ACCOUNT_ID}"
+echo "                        AWS Region: ${AWS_REGION}"
+echo "                           Cluster: ${CLUSTER_NAME}"
+echo "                         SQS Queue: ${SQS_QUEUE_NAME}"
+echo "              EventBridge ASG Rule: ${EVENTBRIDGE_ASG_RULE_NAME}"
+echo "             EventBridge Spot Rule: ${EVENTBRIDGE_SPOT_RULE_NAME}"
+echo " EventBridge EC2 State Change Rule: ${EVENTBRIDGE_EC2_STATE_CHANGE_RULE_NAME}"
+echo "                ASG Lifecycle Hook: ${LIFECYCLE_HOOK_NAME}"
+echo "                        IAM Policy: ${IAM_POLICY}"
+echo "                          IAM Role: ${ROLE_NAME}"
+echo "                          ECR REPO: ${ECR_PLAYGROUND_REPO}"
+echo "                     NTH++ Version: ${VERSION}"
+echo "                          Pod Role: ${POD_ROLE}"
+echo "                   Pod Role Policy: ${POLICY_NAME}"
+echo "                   Service Account: ${SERVICE_ACCOUNT_NAME}"
+echo "                             FLEET: ${FLEET}"
+echo "======================================================================"
+
+echo "sleeping for 20 secs for review of the following parameters, abort w/ CTRL + c"
```

Comment:
5 secs?  

Good CX 👍🏼 .  

thinking out loud: Might be useful to print it out at the end of setup too if the subsequent commands generate a lot of output

Reply:
oops, that was for testing 

That's a good idea, I'll add that.